session-loader/Development/IDE/Session.hs

```diff
@@ -448,7 +437,8 @@ newComponentCache logger exts cradlePath hsc_env uids ci = do
     -- the component, in which case things will be horribly broken anyway.
     -- Otherwise, we will immediately attempt to reload this module which
     -- causes an infinite loop and high CPU usage.
-    let special_target = TargetDetails (mkModuleName "special") targetEnv targetDepends [componentFP ci]
+    let special_target = TargetDetails (TargetModule m) targetEnv targetDepends [componentFP ci]
```

Comment:
This seems strange, though I suppose it predates this PR. Can't we use `TargetFile` now?

Reply:
I'm not sure what's supposed to be the special target. In the original version of the code, it was simply the string "special".

Reply:
I think it might be `_cfp` from `session`, which might need to be passed into `newComponentCache` as an extra argument.

/cc @fendor

Reply:
Yeah, that makes sense. Fixed